llvm/lib/CodeGen/CodeGenPrepare.cpp

```diff
@@ -6986,6 +6987,11 @@ bool CodeGenPrepare::optimizeSelectInst(SelectInst *SI) {
   // Increment the current iterator to skip all the rest of select instructions
   // because they will be either "not lowered" or "all lowered" to branch.
   CurInstIterator = std::next(LastSI->getIterator());
+  // Examine any debug-info attached to those instructions, which won't be seen
+  // elsewhere. But only do it once we return and have potentially rewritten
+  // valeus.
+  for (SelectInst *SI : ArrayRef(ASI).drop_front())
```

Comment:
SGTM

>  But only do it once we return and have potentially rewritten valeus.

nit: typo, and also I am not 100% sure what "once we return" means in this context - is there any way to rewrite this slightly?

Reply:
(new version sounds good, thanks)